ospatch/updates_windows.go

```diff
@@ -185,33 +179,38 @@ func installWUAUpdates(r *patchRun) error {
 		return nil
 	}
 
-	logger.Debugf("DEBUG: %d Windows updates available\n", count)
+	logger.Debugf("%d Windows updates available\n", count)
 
-	class := make(map[string]struct{})
+	classFilter := make(map[string]struct{})
 	excludes := make(map[string]struct{})
 	if r.Job.PatchConfig.WindowsUpdate != nil {
 		for _, c := range r.Job.PatchConfig.WindowsUpdate.Classifications {
 			sc, ok := classifications[c]
 			if !ok {
 				return fmt.Errorf("Unknown classification: %s", c)
 			}
-			class[sc] = struct{}{}
+			classFilter[sc] = struct{}{}
 		}
+		logger.Debugf("Filtering by classifications: %q\n", r.Job.PatchConfig.WindowsUpdate.Classifications)
 
 		for _, e := range r.Job.PatchConfig.WindowsUpdate.Excludes {
 			excludes[e] = struct{}{}
 		}
+		logger.Debugf("Filtering out KBs: %q\n", r.Job.PatchConfig.WindowsUpdate.Excludes)
 	}
 
 	for i := 0; i < int(count); i++ {
+		if err := r.reportContinuingState(osconfigpb.Instance_APPLYING_PATCHES); err != nil {
```

Comment:
What are the chances this leads to DOSing? If that's a possibility, it would be better to do this once before the loop. We can do this at each step when we know it's safe and users want that level of responsiveness.

Reply:
Windows updates are slow to install, I don't see this breaking 1QPS

Reply:
Chatted offline and decided that we should be fine.